hopsworksai/resource_cluster.go

```diff
@@ -28,7 +28,7 @@ func defaultRonDBConfiguration(cloud api.CloudProvider) api.RonDBConfiguration {
 	ronDB := api.RonDBConfiguration{
 		Configuration: api.RonDBBaseConfiguration{
 			NdbdDefault: api.RonDBNdbdDefaultConfiguration{
-				ReplicationFactor: 2,
+				ReplicationFactor: 1,
```

Comment:
Are these change enough to deploy all in one nodes?
Shouldn't there be some check depending of the hopsworks version?

Reply:
I think the check should be on the backend side not on terraform. Also, It shouldn't affect existing users unless if they actively updated their terraform client

Reply:
The check is done in the back-end but I just thought of one combination which is problematic.

Clusters <= 2.5.0 with
* Number of replicas: 1
* Datanodes: 1
* MySQL servers: 1

should create a full RonDB cluster with the specs above and not an all-in-one VM because it is not supported for 2.5 I should fix it in hopsworks-cloud

Reply:
As the fix in hopsworks-cloud been done? Can you point me to it?
Also I am confused how it will affect existing clusters.

Reply:
@berthoug existing users not existing clusters. If you have an older version of terraform and using default configurations, you will create a RonDB cluster with 2 replicas. If the users upgrade their hopsworks provider version then, the default behaviour will be to create an all in one RonDB

Reply:
Yea but what about existing clusters. Won't terraform say that the cluster is malformed and try to redeploy it?
And does that mean that we can't release this until we release 3.0?

Reply:
I don't think it should affect existing clusters since in the terraform side, we don't check on actual nodes created, it is mainly on the backend side @kouzant ? 
Yes, I think we should release this with #141 (enforcing RonDB) as part of a major release v1.0.0

Reply:
No, it does not affect existing clusters. The change is only contained in the backend for new cluster.

I tested it. Launched a cluster with "old" single node configuration with Tf. Switched to this version of Tf with the new defaults and it doesn't change the state.